deepspeed/runtime/utils.py

```diff
@@ -365,42 +365,52 @@ def clip_grad_norm_(parameters, max_norm, norm_type=2, mpu=None):
     parameters = list(filter(lambda p: p.grad is not None, parameters))
     max_norm = float(max_norm)
     norm_type = float(norm_type)
+    all_norms = []
     if norm_type == inf:
-        total_norm = max(p.grad.data.abs().max() for p in parameters)
-        total_norm_cuda = get_accelerator().FloatTensor([float(total_norm)])
+        for p in parameters:
+            all_norms.append(p.grad.data.abs().max().float())
+        total_norm = torch.stack(all_norms).max()
+        origin_device = total_norm.device.type
```

Comment:
Should this not be `total_norm.device`? Won't the current code lose the device index?

Reply:
fixed